src/main/sensors/barometer.c

```diff
@@ -315,54 +327,26 @@ void baroSetGroundLevel(void)
     baroSetCalibrationCycles(SET_GROUND_LEVEL_BARO_CYCLES);
 }
 
-#define PRESSURE_SAMPLES_MEDIAN 3
-
-static int32_t applyBarometerMedianFilter(int32_t newPressureReading)
+#define PRESSURE_SAMPLES_MEDIAN 9
```

Comment:
median filter is slow. 3-point median is great for single outlier removal. 9 taps probably means that you are masking other problem.